src/sys.rs

```diff
@@ -0,0 +1,425 @@
+            pub fn mach_timebase_info(info: *mut mach_timebase_info) -> c_int;
+        }
+
+        pub fn info() -> &'static mach_timebase_info {
+            static mut INFO: mach_timebase_info = mach_timebase_info {
+                numer: 0,
+                denom: 0,
+            };
+            static ONCE: Once = ONCE_INIT;
+
+            unsafe {
+                ONCE.call_once(|| {
+                    mach_timebase_info(&mut INFO);
+                });
+                &INFO
+            }
+        }
+    }
+
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    mod unix {
+        use libc::{c_int, timespec};
+
+        #[cfg(all(not(target_os = "android"),
+                  not(target_os = "bitrig"),
+                  not(target_os = "nacl"),
+                  not(target_os = "openbsd")))]
+        #[link(name = "rt")]
+        extern {}
+
+        extern {
+            pub fn clock_gettime(clk_id: c_int, tp: *mut timespec) -> c_int;
+        }
+    }
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub fn get_time() -> (i64, i32) {
+        use std::ptr;
+        let mut tv = libc::timeval { tv_sec: 0, tv_usec: 0 };
+        unsafe { mac::gettimeofday(&mut tv, ptr::null_mut()); }
+        (tv.tv_sec as i64, tv.tv_usec * 1000)
+    }
+
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    pub fn get_time() -> (i64, i32) {
+        let mut tv = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+        unsafe { unix::clock_gettime(libc::CLOCK_REALTIME, &mut tv); }
+        (tv.tv_sec as i64, tv.tv_nsec as i32)
+    }
+
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub fn get_precise_ns() -> u64 {
+        unsafe {
+            let time = mac::mach_absolute_time();
+            let info = mac::info();
+            time * info.numer as u64 / info.denom as u64
+        }
+    }
+
+    #[cfg(all(not(target_os = "macos"), not(target_os = "ios")))]
+    pub fn get_precise_ns() -> u64 {
+        let mut ts = libc::timespec { tv_sec: 0, tv_nsec: 0 };
+        unsafe {
+            unix::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
+        }
+        (ts.tv_sec as u64) * 1000000000 + (ts.tv_nsec as u64)
+    }
+}
+
+#[cfg(windows)]
+#[allow(non_snake_case)]
+mod inner {
+    use libc::{self, WORD, DWORD, LONG, WCHAR};
+    use ::{Tm};
+    use std::sync::{Once, ONCE_INIT};
+
+    pub fn frequency() -> libc::LARGE_INTEGER {
+        static mut FREQUENCY: libc::LARGE_INTEGER = 0;
+        static ONCE: Once = ONCE_INIT;
+
+        unsafe {
+            ONCE.call_once(|| {
+                libc::QueryPerformanceFrequency(&mut FREQUENCY);
+            });
+            FREQUENCY
+        }
+    }
+
+    #[repr(C)]
+    #[derive(Default)]
+    struct SystemTime {
+        wYear: WORD,
+        wMonth: WORD,
+        wDayOfWeek: WORD,
+        wDay: WORD,
+        wHour: WORD,
+        wMinute: WORD,
+        wSecond: WORD,
+        wMilliseconds: WORD,
+    }
```

Comment:
Do you mean [this winapi](https://crates.io/crates/winapi)? We can. The types and functions used in this module seem to be [unimplemented](https://github.com/retep998/winapi-rs/blob/master/lib/kernel32-sys/src/lib.rs#L369).